build_scripts/job/dreambooth/utils/gen_utils.py

```
@@ -0,0 +1,259 @@
+        pbar: mytqdm = None
+):
+    """
+
+    @param args: A DreamboothConfig
+    @param class_gen_method
+    @param accelerator: An optional existing accelerator to use.
+    @param ui: Whether this was called by th UI, or is being run during training.
+    @param pbar: Progress bar to use.
+    @return:
+    generated: Number of images generated
+    images: A list of images or image paths, depending on if returning to the UI or not.
+    if ui is False, this will return a second array of paths representing the class paths.
+
+    """
+    out_images = []
+    instance_prompts = []
+    class_prompts = []
+    try:
+        status.textinfo = "Preparing dataset..."
+        prompt_dataset = ClassDataset(
+            args.concepts(), args.model_dir, args.resolution, False, args.disable_class_matching, pbar=pbar
+        )
+        instance_prompts = prompt_dataset.instance_prompts
+        class_prompts = prompt_dataset.class_prompts
+    except Exception as p:
+        print(f"Exception generating dataset: {str(p)}")
+        traceback.print_exc()
+        if ui:
+            return 0, []
+        else:
+            return 0, instance_prompts, class_prompts
+
+    set_len = prompt_dataset.__len__()
+    if set_len == 0:
+        print("Nothing to generate.")
+        if ui:
+            return 0, []
+        else:
+            return 0, instance_prompts, class_prompts
+
+    print(f"Generating {set_len} class images for training...")
+    status_handler = None
+    if pbar is None:
+        logging.getLogger(__name__).info("Creating new progress bar")
+        pbar = mytqdm(total=set_len, desc=f"Generating class images 0/{set_len}:", position=0, target="dreamProgress")
+    else:
+        logging.getLogger(__name__).info("Using existing progress bar")
+        pbar.reset(total=set_len)
+        if getattr(pbar, "user", None):
+            try:
+                status_handler = StatusHandler(user_name=pbar.user, target="dreamProgress")
+            except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```

Similar issue at line number 203.